checker/closure.ml

```diff
@@ -456,13 +454,10 @@ let rec to_constr constr_fun lfts v =
     | FFlex (ConstKey op) -> Const op
     | FInd op -> Ind op
     | FConstruct op -> Construct op
-    | FCase (ci,p,c,ve) ->
-	Case (ci, constr_fun lfts p,
-              constr_fun lfts c,
-	      Array.map (constr_fun lfts) ve)
-    | FCaseT (ci,p,c,ve,e) -> (* TODO: enable sharing, cf FCLOS below ? *)
-        to_constr constr_fun lfts
-	  {norm=Red;term=FCase(ci,mk_clos2 e p,c,mk_clos_vect e ve)}
+    | FCaseT (ci,p,c,ve,e) ->
+       let fp = mk_clos2 e p in
```

Comment:
It seems we use `mk_clos` in the kernel here. Any idea when we should use `mk_clos` vs `mk_clos2`. By the way, I don't understand what it brings to the world to maintain to copies of this code. Selling it as more trust is just a lie.

Reply:
It changes the depth of encoding. I believe it has an effect on performances, but I must admit I am pretty clueless whether this really matters.

Reply:
Yeah, my point was that I don't see a reason why the better version would not be the same in the kernel and the checker.